_drafts/2017-09-01-first-review-experiences.md

```diff
@@ -0,0 +1,137 @@
+by testing core functionality of the package by running the examples
+that were mentioned in the README of the package. I think this is a good
+starting point because you get a feeling of what the author wants to
+achieve with the package. Later on during the review I came up with my
+own queries (side note: this R package interacts with an API from which
+you can query patents). During the review I used to switch between 
+writing queries like a normal user of the package
+would do and checking the code. When I saw something in the code that
+wasn't quite clear to me or looked wrong I went back to writing new
+queries to check whether the behavior of the methods was as expected.
+With this approach I was able to give feedback to the package author
+which led to the inclusion of an additional unit test, a helper function
+that makes the package easier to use, clarification of an error message
+and an improved documentation. You can find the review I did [here](https://github.com/ropensci/onboarding/issues/112).
+
+In the rest of this article you can read what I gained from doing a
+review personally.
+
+Contributing to the open source community
+-----------------------------------------
+
+When people think about contributing to the open source community, the
+first thought is about creating a new R package or contributing to one
+of the major packages out there. But not everyone has the resources
+(e.g. time) to do so. You also don't have awesome ideas every other day
+which can immediately be implemented into new R packages to be used by
+the community. Besides contributing with code there are also lots of
+other things than can be useful for other R users, for example writing
+blog posts about problems you solved, speaking at meetups or reviewing
+code to help improve it. What I like much about reviewing code is that
+people see things differently and have other experiences. As a reviewer,
+you see a new package from the user's perspective which can be hard for
+the programmer themselves. Having someone else
+review your code helps finding things that are missing because they seem
+obvious to the package author or detect code pieces that require more
+testing. I had a great feeling when I finished the review, since I had
+helped improve an already amazing R package a little bit more.
+
+While writing this blog post I found a nice article about [contributing
+to the tidyverse](http://www.tidyverse.org/articles/2017/08/contributing/) which is
+mostly also applicable to other R packages in my opinion.
+
+Reviewing helps improving your own coding style
+-----------------------------------------------
+
+When I write R code I usually try to do it in the best way possible.
+[Google's R Style Guide](https://google.github.io/styleguide/Rguide.xml)
+is a good start to get used to coding best practice in R and I also
+enjoyed reading [Programming Best Practices
+Tidbits](https://github.com/thomasdavis/best-practices). So normally
```

Comment:
I see this https://github.com/thomasdavis/best-practices is forked from https://github.com/timoxley/best-practices.
I'm naive about this stuff, but wanted to check that you used the correct link. The timoxley one looks more recently updated